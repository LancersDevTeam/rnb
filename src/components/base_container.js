// @flow
import React, { Component } from 'react'
import { StatusBar } from 'react-native'
import { Container as NBContainer } from 'native-base'
import styles from './base_container_style'

type State = any
type Props = any

export class BaseContainer extends Component<Props, State> {
  render() {
    return (
      <NBContainer {...this.props} style={styles.container}>
        <StatusBar />
        {this.props.children}
      </NBContainer>
    )
  }
}
