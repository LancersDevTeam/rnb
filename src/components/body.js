// @flow
import React, { Component } from 'react'
import { Body as NBBody } from 'native-base'

type State = any
type Props = any

export class Body extends Component<Props, State> {
  render() {
    return <NBBody {...this.props}>{this.props.children}</NBBody>
  }
}
