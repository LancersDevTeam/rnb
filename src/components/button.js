// @flow
import React, { Component } from 'react'

import { Button as NBButton } from 'native-base'

type State = any
type Props = any

export class Button extends Component<Props, State> {
  render() {
    return (
      <NBButton rounded block {...this.props}>
        {this.props.children}
      </NBButton>
    )
  }
}
