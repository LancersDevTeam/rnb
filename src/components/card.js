// @flow
import React, { Component } from 'react'
import { Card as NBCard } from 'native-base'

type State = any
type Props = any

export class Card extends Component<Props, State> {
  render() {
    return <NBCard {...this.props}>{this.props.children}</NBCard>
  }
}
