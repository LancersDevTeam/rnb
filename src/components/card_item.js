// @flow
import React, { Component } from 'react'

import { CardItem as NBCardItem } from 'native-base'

type State = any
type Props = any

export class CardItem extends Component<Props, State> {
  _root: any

  setNativeProps = (nativeProps: any) => {
    this._root.setNativeProps(nativeProps)
  }

  render() {
    return (
      <NBCardItem ref={component => (this._root = component)} {...this.props}>
        {this.props.children}
      </NBCardItem>
    )
  }
}
