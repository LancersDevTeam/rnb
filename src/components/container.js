// @flow
import React, { Component } from 'react'
import { Container as NBContainer } from 'native-base'

type State = any
type Props = any

export class Container extends Component<Props, State> {
  render() {
    return <NBContainer {...this.props}>{this.props.children}</NBContainer>
  }
}
