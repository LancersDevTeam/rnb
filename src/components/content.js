// @flow
import React, { Component } from 'react'
import { Content as NBContent } from 'native-base'

type State = any
type Props = any

export class Content extends Component<Props, State> {
  render() {
    return <NBContent {...this.props}>{this.props.children}</NBContent>
  }
}
