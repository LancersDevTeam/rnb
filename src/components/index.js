export * from './base_container'
export * from './body'
export * from './button'
export * from './card'
export * from './card_item'
export * from './container'
export * from './content'
export * from './deck_swiper'
export * from './form'
export * from './header'
export * from './icon'
export * from './input'
export * from './input_group'
export * from './item'
export * from './label'
export * from './left'
export * from './list'
export * from './list_item'
export * from './right'
export * from './separator'
export * from './switch'
export * from './text'
export * from './thumbnail'
