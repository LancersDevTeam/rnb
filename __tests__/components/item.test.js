// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import { Item } from 'app/components'

it('renders correctly', () => {
  const tree = renderer.create(<Item />).toJSON()
  expect(tree).toMatchSnapshot()
})
