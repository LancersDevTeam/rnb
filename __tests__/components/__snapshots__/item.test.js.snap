// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderBottomWidth": 1,
      "borderColor": "#D9D5DC",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "marginLeft": 2,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
/>
`;
